baremetal/metal3machine_manager_test.go

```diff
@@ -3517,129 +3528,312 @@ var _ = Describe("Metal3Machine manager", func() {
+					{
+						TypeMeta: metav1.TypeMeta{},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test2",
+						},
+					},
+				},
+			},
+			expectedMachineSet: nil,
+			expectError:        true,
+			expectedMD:         false,
+			expectedMDName:     "",
+		}),
+		Entry("Should not find the expected MachineDeployment name, MachineSet OwnerRef APIVersion is not correct", testCaseGetMachineDeploymentName{
+			Machine: machineOwnerRefToMachineSet(),
+			MachineSetList: &capi.MachineSetList{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: capi.GroupVersion.String(),
+					Kind:       "MachineSet",
+				},
+				ListMeta: metav1.ListMeta{},
+				Items: []capi.MachineSet{
+					{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: capi.GroupVersion.String(),
+							Kind:       "MachineSet",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "ms-test1",
+							OwnerReferences: []metav1.OwnerReference{
+								{
+									APIVersion: capm3.GroupVersion.String(),
+									Kind:       "MachineDeployment",
+									Name:       "test1",
+								},
+							},
+						},
+					},
+					{
+						TypeMeta: metav1.TypeMeta{},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test2",
+						},
+					},
+				},
+			},
+			expectedMachineSet: nil,
+			expectError:        true,
+			expectedMD:         false,
+			expectedMDName:     "",
+		}),
 	)
 
 	type testCaseGetMachineSet struct {
 		Machine            *capi.Machine
-		MachineSets        []*capi.MachineSet
+		MachineSetList     *capi.MachineSetList
 		expectedMachineSet *capi.MachineSet
 		expectError        bool
 	}
 	DescribeTable("Test GetMachineSet",
 		func(tc testCaseGetMachineSet) {
 			objects := []client.Object{}
-			for _, ms := range tc.MachineSets {
-				objects = append(objects, ms)
+			if tc.MachineSetList != nil {
+				for i := range tc.MachineSetList.Items {
+					objects = append(objects, &tc.MachineSetList.DeepCopy().Items[i])
+				}
 			}
+			if tc.Machine != nil {
+				objects = append(objects, tc.Machine)
+			}
+
 			c := fakeclient.NewClientBuilder().WithScheme(setupSchemeMm()).WithObjects(objects...).Build()
 			machineMgr, err := NewMachineSetManager(c, tc.Machine,
-				tc.MachineSets, nil, logr.Discard(),
+				tc.MachineSetList,
+				logr.Discard(),
 			)
 			Expect(err).NotTo(HaveOccurred())
 
 			result, err := machineMgr.getMachineSet(context.TODO())
 			if tc.expectError {
 				Expect(err).To(HaveOccurred())
+				Expect(tc.expectedMachineSet).To(BeNil())
 			} else {
-				Expect(err).NotTo(HaveOccurred())
-			}
-			machineSetObjects := capi.MachineSetList{}
-
-			for ms := range machineSetObjects.Items {
-				tc.expectedMachineSet = &machineSetObjects.Items[ms]
-				Expect(result).To(Equal(tc.expectedMachineSet))
+				for i := range tc.MachineSetList.Items {
+					foundMS := &tc.MachineSetList.Items[i]
+					key := client.ObjectKey{
+						Name:      foundMS.Name,
+						Namespace: foundMS.Namespace,
+					}
+					err = c.Get(context.TODO(), key, foundMS)
```

Comment:
Again, here, I'm not sure we're testing anything by running `c.Get` on the fakeclient?

Reply:
Please check above comment